File: apps/backend/app/schemas/deduction.py
from datetime import datetime
from typing import List, Optional

from pydantic import BaseModel, Field


class DeductionCreateRequest(BaseModel):
    category: str  # damage | cleaning | unpaid_rent | other
    description: str
    amount: float = Field(gt=0)


class DeductionResponse(BaseModel):
    id: str
    org_id: str
    lease_id: str
    tenant_id: str
    category: str
    description: str
    amount: float
    evidence_keys: List[str] = []
    approved_by: Optional[str] = None
    created_at: datetime
    updated_at: datetime


class DeductionSummary(BaseModel):
    items: List[DeductionResponse]
    total: float

File: apps/frontend/src/types/deduction.ts
export type DeductionCategory = 'damage' | 'cleaning' | 'unpaid_rent' | 'other'

export interface Deduction {
  id: string
  org_id: string
  lease_id: string
  tenant_id: string
  category: DeductionCategory
  description: string
  amount: number
  evidence_keys: string[]
  approved_by?: string
  created_at: string
  updated_at: string
}

export interface DeductionSummary {
  items: Deduction[]
  total: number
}

export interface DeductionCreateRequest {
  category: DeductionCategory
  description: string
  amount: number
}
